### PR TITLE
Refactor Project Services and Routes

### LINKED ISSUE #1631: Wrong namespace set in `garden` cluster kubeconfig on MY ACCOUNT page
**What happened**:
When you open the 'MY ACCOUNT' page on Project A and then switch to Project B, the namespace of Project A will be present in the kubeconfig. This can be observed either when downloading the kubeconfig or when viewing it on the Preview tab on the 'MY ACCOUNT' page.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know**:

**Environment**:

### PR DESCRIPTION
**Pull request by petersutter**
**What this PR does / why we need it**:
This PR includes several changes aimed at improving the project services and routes within the application. The key changes are:
- Renamed the legacy project service to `namespace`.
- Added a new route for projects and removed the old from/to mapping.
- Updated the application to use the new projects route and removed the redundant from/to mapping code.
- Improved the handling of project deletions.
- Extracted cost-related logic into a separate composable object.
- Made minor improvements to the codebase.

**Which issue(s) this PR fixes**:
Fixes #1631

**Special notes for your reviewer**:

**Release note**:
```other developer

```

### CODE REVIEW

File: frontend/src/views/GMembers.vue [hunk 1]
```diff
@@ -638,8 +638,8 @@ async function onRemoveUser ({ username }) {
   }
   await memberStore.deleteMember(username)
   if (isCurrentUser(username) && !isAdmin.value) {
-    if (projectList.value.length > 0) {
-      const p1 = projectList.value[0]
+    if (projectsNotMarkedForDeletion.value.length > 0) {
+      const p1 = projectsNotMarkedForDeletion.value[0]
```
Comment (holgerkoser) [inline frontend/src/views/GMembers.vue:642]:
```js
      const project = projectsNotMarkedForDeletion.value[0]
      const namespace = project.spec.namespace
      await router.push({ name: 'ShootList', params: { namespace } })
```

File: backend/lib/routes/index.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ module.exports = {
   '/cloudprofiles': require('./cloudprofiles'),
   '/seeds': require('./seeds'),
   '/gardenerextensions': require('./gardenerExtensions'),
-  '/namespaces': require('./namespaces'),
+  '/namespaces': require('./namespaces'), // TODO delete this legacy route
```
[the remark this answers is not in the document]
Comment (petersutter) [reply backend/lib/routes/index.js:21]: ✅

File: frontend/src/views/GMembers.vue (hunk 1 above)
Comment (petersutter) [reply frontend/src/views/GMembers.vue:642]: nice find
✅

**Review by grolu — APPROVED** [review]
/lgtm

**Review by holgerkoser — APPROVED** [review]
/lgtm

File: frontend/src/views/GAdministration.vue [hunk 3]
```diff
@@ -571,13 +571,18 @@ const {
...
   shootCustomFields,
   projectOwner: owner,
-  projectCostObject: costObject,
   projectCreationTimestamp: creationTimestamp,
   projectCreatedBy: createdBy,
   projectDescription: description,
   projectPurpose: purpose,
   projectStaleSinceTimestamp: staleSinceTimestamp,
   projectStaleAutoDeleteTimestamp: staleAutoDeleteTimestamp,
+  costObject,
```
[the remark this answers is not in the document]
Comment (holgerkoser) [reply frontend/src/views/GAdministration.vue:581]: We should talk about this tomorrow. But for now I will makr it as resolved

**Pull request merged**